subprojects/core/src/main/java/org/gradle/api/tasks/WriteProperties.java

```diff
@@ -191,30 +196,58 @@ public void setEncoding(String encoding) {
      * Returns the output file to write the properties to.
      */
     @OutputFile
+    @Optional
+    @Deprecated
     public File getOutputFile() {
-        return getServices().get(FileOperations.class).file(outputFile);
+        DeprecationLogger.deprecateProperty(getClass(), "outputFile").replaceWith("destinationFile")
+            .willBeRemovedInGradle9()
+            .withDslReference()
+            .nagUser();
```

Comment:
:x: fix the link, see comment above, and extract deprecation logging in a common function